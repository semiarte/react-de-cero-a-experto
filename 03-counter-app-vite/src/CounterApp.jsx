import { useState } from "react";
import PropTypes from 'prop-types'

// eslint-disable-next-line react/prop-types
const CounterApp = ({ title, value }) => {

    const [ counter, setCounter ] = useState( value );
    
    const handleApp = () => {
        // console.log(event);
        setCounter ( counter + 1 );
        // setCounter( (c) => c + 1 );
    }

    const handleSubstract = () => {
        setCounter ( counter - 1 );
    }

    const reset = () => {
        setCounter (value);
    }

    return (
        <div className="mainContent">
            <h1>{ title }</h1>
            <h2>Counter: { counter }</h2>

            <button onClick={ handleApp }>+1</button>
            <button onClick={ handleSubstract }>-1</button>
            <button onClick={ reset }>reset</button>
        </div>
    );
}

CounterApp.propTypes = {
    title: PropTypes.string.isRequired,
    value: PropTypes.number
}

export default CounterApp;