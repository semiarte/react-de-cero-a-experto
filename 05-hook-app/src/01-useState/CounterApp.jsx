import { useState } from "react"

export const CounterApp = () => {
    // el estado inicial es un objeto
    const [state, setCounter] = useState({
        counter1: 10,
        counter2: 20,
        counter3: 30,
    });

    const {counter1, counter2, counter3} = state;

    return(
        <>
            <h1 key="1">{counter1}</h1>
            <h1 key="2">{counter2}</h1>
            <h1 key="3">{counter3}</h1>

            <hr />

            <button className="btn" onClick={
                () => setCounter({
                    // preservamos los valores de state y mutamos counter1
                    ... state,
                    counter1: counter1 + 1,
                })
            }>+1</button>
        </>
    )
}