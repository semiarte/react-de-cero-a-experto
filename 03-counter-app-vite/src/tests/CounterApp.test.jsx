/* eslint-disable no-undef */
import { render } from '@testing-library/react';
import CounterApp from '../CounterApp';

describe('Pruebas en el <CounterApp />', () => {
    const InitialValue = 10;
    
    test('debe de hacer match con el snapshot', () => { 
        const { container } = render(<CounterApp title='Counter' value={InitialValue} />)
        expect(container).toMatchSnapshot();
    });
    
})