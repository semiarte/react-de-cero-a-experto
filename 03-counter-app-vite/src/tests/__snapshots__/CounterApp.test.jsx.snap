// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <CounterApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="mainContent"
  >
    <h1>
      Counter
    </h1>
    <h2>
      Counter: 
      10
    </h2>
    <button>
      +1
    </button>
    <button>
      -1
    </button>
    <button>
      reset
    </button>
  </div>
</div>
`;
